### PR TITLE
Multiple cluster support.

### PR DESCRIPTION
**Pull request by jarekwg**
Added support for running multiple clusters, with the expectation there'll be one gitops instance running on each.

- Renamed 'Cluster' to 'AppDefinitions' (a bit wordy, but hopefully more descriptive, and disambiaguates from actual kube cluster). Alternative name could've also been `ClusterRepo`, but I felt that still had some ambiguity to it.. ;/
- kubeconfig file path can now be passed to invoke commands, to deploy gitops server to other clusters easily.
- Made gitops bail out if it the app it's looking at isn't targeting the cluster it's running on.

### CODE REVIEW

File: tests/utils.py [hunk 1]
```diff
@@ -46,3 +47,39 @@ def create_test_yaml(fg=4, bg=2):
     with open('/tmp/secrets.yml', 'w+') as fh:
         fh.write(yaml.dump(data))
     return '/tmp/'
+
+
+# def create_test_kubeconfig():
```
Comment (Sticky-Bits) [inline tests/utils.py:52]: Killing this? Keeping it?

Comment (jarekwg) [reply tests/utils.py:52]: I want to use it, but the way I've defined the reading from kubeconfig currently (for cluster name, in init) means i can't get this function to run early enough during tests. :D
Might change that later..
I think we'll still want to use this dummy kube config when we add more tests in future.
